core/src/com/github/czyzby/tests/reflected/widgets/CodeTextArea.java

```diff
@@ -0,0 +1,44 @@
+package com.github.czyzby.tests.reflected.widgets;
+
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.kotcrab.vis.ui.widget.VisTextArea;
+
+/**
+ * Custom {@link VisTextArea} used to display LML code in main view. Supports embedding in scrollpane
+ * by calcuating required space needed for current text.
+ * @author Kotcrab
+ */
+public class CodeTextArea extends VisTextArea {
+    private GlyphLayout prefSizeLayout;
+    private String lastText;
+
+    public CodeTextArea(String text, String styleName) {
+        super(text, styleName);
+    }
+
+    @Override
+    protected void initialize() {
+        super.initialize();
+        prefSizeLayout = new GlyphLayout();
+    }
+
+    @Override
+    public float getPrefHeight() {
+        updatePrefSizeLayoutIfNeeded();
+        return prefSizeLayout.height + super.getPrefHeight();
+    }
+
+    @Override
+    public float getPrefWidth() {
+        updatePrefSizeLayoutIfNeeded();
+        return prefSizeLayout.width + super.getPrefHeight();
+    }
+
+    private void updatePrefSizeLayoutIfNeeded() {
+        final String text = getText();
+        if (lastText != text) {
```

Comment:
Yes `getText()` always returns same object if nothing was changed and here we don't really care if actual text content is the same. Strings are immutable so if it changes we will get different object reference.